packages/forklift-console-plugin/src/modules/Providers/data.ts

```diff
@@ -1,68 +1,17 @@
 import { useMemo } from 'react';
-import * as C from 'src/utils/constants';
 import { useProviders } from 'src/utils/fetch';
-import { groupVersionKindForObj } from 'src/utils/resources';
-import { ProviderStatus } from 'src/utils/types';
 
 import { useInventoryProvidersQuery } from '@kubev2v/legacy/queries';
 import {
   IOpenShiftProvider,
   IOpenStackProvider,
-  IProviderObject,
   IProvidersByType,
   IRHVProvider,
   IVMwareProvider,
 } from '@kubev2v/legacy/queries/types';
-import { V1beta1Provider, V1beta1ProviderStatusConditions } from '@kubev2v/types';
-import { K8sGroupVersionKind, OwnerReference } from '@openshift-console/dynamic-plugin-sdk';
+import { V1beta1Provider } from '@kubev2v/types';
 
-const conditionState = (state: string) =>
-  state === 'True' || state === 'False' ? state : 'Unknown';
-
-interface SupportedCondition {
-  status: string;
-  message: string;
-}
-
-interface PositiveConditions {
-  Ready?: SupportedCondition;
-  Validated?: SupportedCondition;
-  ConnectionTestSucceeded?: SupportedCondition;
-  InventoryCreated?: SupportedCondition;
-  LoadInventory?: SupportedCondition;
-}
-interface NegativeConditions {
-  URLNotValid?: SupportedCondition;
-  ProviderTypeNotSupported?: SupportedCondition;
-  SecretNotValid?: SupportedCondition;
-  SettingsNotValid?: SupportedCondition;
-  ConnectionTestFailed?: SupportedCondition;
-}
-
-export type SupportedConditions = PositiveConditions & NegativeConditions;
-
-export interface MergedProvider {
-  [C.NAME]: string;
-  [C.NAMESPACE]: string;
-  [C.URL]: string;
-  [C.TYPE]: string;
-  [C.UID]: string;
-  [C.SECRET_NAME]: string;
-  [C.GVK]: K8sGroupVersionKind;
-  [C.DEFAULT_TRANSFER_NETWORK]: string;
-  [C.CLUSTER_COUNT]: number;
-  [C.HOST_COUNT]: number;
-  [C.VM_COUNT]: number;
-  [C.NETWORK_COUNT]: number;
-  [C.STORAGE_COUNT]: number;
-  [C.PHASE]: ProviderStatus;
```

Comment:
this is an example when we loose type information - the generated API use only simple primitive types which is only a subset of TS capabilities